### DISCUSSION TITLE
How to use object xml to give objects scene names?

### DISCUSSION THREAD

**Discussion by jrounds** [NVlabs/sionna#221] [General]
Hi all,

Maybe someone knows this.

If have an xml with object files that correctly load in sionna, what do I add to the xml to give scene.get("...") a key that I know by inspect the xml?

For example suppose this is an object in a valid scene xml:
```
#... snip ...
    <shape type="obj">
        <boolean name="face_normals" value="true"/>
        <string name="filename" value="my_wall.obj" />
        <ref id="mat-my_material_name" name="bsdf"/>
    </shape>
```

This loads, but is in sionna's scene as "_unnamed-##".  I want to assign it a name so I can scene.get("my_wall") or something like that.

I assume the xml can be altered to make it have an object title or tag or something that can be found. TBH I have iterated scene objects and the best hint you get to the original object is the material name.

**Comment by merlinND** [comment]
Hello @jrounds,

Does adding an `id="name"` property work for your?

```xml
<shape type="obj" id="myname">
    ...
</shape>
```

**Reply by jrounds** [reply]
That worked perfectly.  Thanks so much!
